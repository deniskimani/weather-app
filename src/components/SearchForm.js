/* eslint-disable consistent-return */
function SearchForm({ searchText, setSearchText, onsubmit }) {
  const handleInputChange = (event) => setSearchText(event.target.value);

  return (
    <div className="search-form">
      <input type="text" value={searchText} onChange={handleInputChange} />
      <button type="submit" onClick={onsubmit}>
        Search
      </button>
    </div>
  );
}
export default SearchForm;
