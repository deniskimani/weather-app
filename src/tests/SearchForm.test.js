import { render } from "@testing-library/react";

import SearchForm from "../components/SearchForm";

describe("App", () => {
  it("renders App component correctly", () => {
    const { asFragment } = render(<SearchForm />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("renders the correct city and location props", () => {
    const { getByText } = render(<SearchForm />);

    expect(getByText("Search")).toBeInstanceOf(HTMLButtonElement);
  });
});
