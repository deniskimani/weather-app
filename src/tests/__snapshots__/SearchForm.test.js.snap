// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders App component correctly 1`] = `
<DocumentFragment>
  <div
    class="search-form"
  >
    <input
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Search
    </button>
  </div>
</DocumentFragment>
`;
